### ISSUE TITLE
tape.gradient slow with RNN GRU model

### ISSUE THREAD

**Issue by ofiryaish** [tensorflow/tensorflow#42990]
I have a trained  keras model, I'm trying to find the gradient of `target_preds` with respect of `sample_inputs` in the trained model. I tried to do it with CNN model and it computed in couple of seconds. When I insert a model that is RNN, the function it take it seems that the same code executes for long time (couple of minutes), and when I interrupt the code- it (is running) at line   `grads = tape.gradient(target_preds, sample_inputs)`).

I'm using Tensorflow 2.3.0

The relevant code is below:

```
    with tf.GradientTape() as tape:
        tape.watch(sample_inputs)
        preds = model(sample_inputs)
        if (target_range is None):
            target_preds = preds[:, :]
        else:
            target_preds = preds[:, target_range]
    
    if(jacobian):
        grads = tape.jacobian(target_preds, sample_inputs)
    else:
        grads = tape.gradient(target_preds, sample_inputs)
    return grads
```

One more important note:
when I used tensorflow 1 I tried the apply the same operation to the same model using: 
```
...
gradients = model.optimizer.get_gradients(model.output[:, c], model.input)
...
get_gradients = keras.backend.function(inputs=input_tensors, outputs=gradients)
...
gradients = get_gradients(_input)
```
This way it was computed in seconds,  but Tensorflow 2 stopped supporting this.

**Comment by ravikyram** [comment]
@ofiryaish 

Looks like code is incomplete. Can you please share colab link or simple standalone code with supporting files to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by ofiryaish** [comment]
I think I found the problem, and it's not a technical one (even though it's really true that performing gradient for GRU is much slower than performing for CNN). It's just much more faster to perform gradient on a batch rather performing on singles, and I didn't take the advantage of this, and therefore it took hours to complete. 

So in overall, you can close the topic. By the way, If you got, in general, a tip for making the gradient procedure faster (maybe using function?) , I would be happy to hear.

thank you

**Comment by ravikyram** [comment]
@ofiryaish 

I am closing the issue as per your comments.Thanks!
